requirements-dev.txt

```diff
@@ -0,0 +1,21 @@
+black
+check-manifest
+flake8
+flake8-builtins
+flake8-comprehensions
+flake8-mutable
+flake8-print
+isort
+mypy
+numpydoc
+pre-commit
+pylint
+pytest
+pytest-cov
+pytest-flake8
+pytest-xdist
+setuptools_scm
+sphinx
+sphinx-automodapi
+twine
+wheel
```

Comment:
After this is merged I'll add a few tests that will use these req deps. Most of them is to check the tarball and inconsistencies that will make the releases more robust or, at least, easier to debug.

Reply:
Cool! Thanks! 

Question before I merge this: I had to specify sphinx-automodapi and numpydoc because I couldn't get readthedocs to build correctly without them in the requirements.txt file (from what I found it seems that allowing specifying requirements under docs started in v2.0, but RTD didn't support that version then). I could look into if that has changed. But should we try to keep these two in the requirements.txt for now?

Reply:
We can added them as extras-docs in the `setup.py`. I'm not sure how RTD works but that should do it. Let me check...